### ISSUE TITLE
Provide guidance how to tune badger.Options

### ISSUE THREAD

**Issue by stephane-martin** [dgraph-io/badger#109]
Hello,

the documentation about Options is a bit terse...

Let's suppose I know:
- the average key size K of the objects
- the average value size V of the objects
- the expected maximum number N of objects to store
- the max amount of RAM R that bagder is supposed to consume

Can i deduct from that information some parameters in Options ?

Thanks!

**Comment by GwynethLlewelyn** [comment]
Also, will tweaking those parameters 'destroy' an existing database? I'm just asking because I have a database with 9 million entries which takes about 3 minutes to load. Doing that all the time the parameters are tweaked might not be a joy :(

**Comment by deepakjois** [comment]
@GwynethLlewelyn Tweaking these options shouldn’t have an effect on an existing Badger datastore. If that happens, we would treat it as a bug.

However, do note that we are still in a pre v1.0 phase. We are making changes aggressively and some changes will break existing datastores (for example see #201). We may also add or remove some options before we stabilize for a v1.0 release.

**Comment by manishrjain** [comment]
So, replying to the original issue, it's hard to calculate the amount of RAM that Badger would take. We checked in changes some time ago to reduce any excessive RAM usage that we were doing unwillingly.

I've updated the godoc for options, so it's clear which options are mandatory, which should be reviewed, which ignored etc. So, have another look at the godoc, and see if you still need more guidance.
